table/tables/tables.go

```diff
@@ -1091,6 +1144,15 @@ func (t *TableCommon) RemoveRecord(ctx sessionctx.Context, h kv.Handle, r []type
 		return err
 	}
 
+	sessVars := ctx.GetSessionVars()
+	sc := sessVars.StmtCtx
+	if sessVars.EnableMutationChecker {
+		if err = CheckDataConsistency(txn, sessVars, t, nil, r, memBuffer, sh); err != nil {
+			return errors.Trace(err)
+		}
+	}
+	memBuffer.Release(sh)
```

Comment:
Will the staging buffer `sh` contain new content? If not we could discard it.

Reply:
> Will the staging buffer sh contain new content? 

No.
What did you mean by discard? Did you mean `MemBuffer.Cleanup`?

Reply:
There is a defer cleanup.